service-base/src/main/java/org/eclipse/hono/service/tenant/BaseTenantService.java

```diff
@@ -194,61 +193,76 @@ private void reply(final Message<JsonObject> request, final AsyncResult<TenantRe
+                    return false;
+                }
+                if (adapters.stream().filter(adapterDetails -> ((JsonObject) adapterDetails).getString(FIELD_ADAPTERS_TYPE) == null)
+                        .findFirst().isPresent()) {
+                    return false;
+                }
             }
+        } catch(ClassCastException e) {
+            // if we did not find a JsonArray or the elements were not of type JsonObject
+            return false;
         }
         return true;
     }
 
+    /**
+     * Get the payload from the request and add default values for not provided optional elements.
+     * <p>
+     * Payload should be checked for validity first, there is no error handling inside this method anymore.
+     * </p>
+     *
+     * @param requestWithCheckedPayload The request containing the payload that was checked for validity already.
+     */
+    private JsonObject getRequestPayload(final JsonObject requestWithCheckedPayload) {
+        final JsonObject payload = requestWithCheckedPayload.getJsonObject(TenantConstants.FIELD_PAYLOAD, new JsonObject());
+        addNotPresentFieldsWithDefaultValuesForTenant(payload);
+        return payload;
+    }
+
     /**
      * Add default values for optional fields that are not filled in the payload.
-     * @param payload The payload to add optional fields to.
-     * @throws ClassCastException If the {@link TenantConstants#FIELD_ADAPTERS_TYPE} element is not a {@link JsonArray}
-     *       or contains elements that are not of type {@link JsonObject}.
+     * <p>
+     * Payload should be checked for validity first, there is no error handling inside this method anymore.
+     * </p>
+     *
+     * @param checkedPayload The checked payload to add optional fields to.
      */
-    private void addNotPresentFieldsWithDefaultValuesForTenant(final JsonObject payload) {
-        if (!payload.containsKey(TenantConstants.FIELD_ENABLED)) {
+    private void addNotPresentFieldsWithDefaultValuesForTenant(final JsonObject checkedPayload) {
+        if (!checkedPayload.containsKey(TenantConstants.FIELD_ENABLED)) {
             log.trace("adding 'enabled' key to payload");
-            payload.put(TenantConstants.FIELD_ENABLED, Boolean.TRUE);
+            checkedPayload.put(TenantConstants.FIELD_ENABLED, Boolean.TRUE);
         }
 
-        final JsonArray adapters = payload.getJsonArray(TenantConstants.FIELD_ADAPTERS);
-        if (adapters != null) {
-            adapters.forEach(elem -> addNotPresentFieldsWithDefaultValuesForAdapter((JsonObject) elem));
+        try {
+            final JsonArray adapters = checkedPayload.getJsonArray(TenantConstants.FIELD_ADAPTERS);
+            if (adapters != null) {
+                adapters.forEach(elem -> addNotPresentFieldsWithDefaultValuesForAdapter((JsonObject) elem));
+            }
+        } catch(ClassCastException e) {
+            // do intentionally nothing, since the payload should have been checked before calling this method
```

Comment:
then you do not need to catch it at all FMPOV

Reply:
Yes, good point. Superfluous code, will remove it.